#### Test how much time the model needs to load etc.

In [1]:
import torch
from model import UNet

In [2]:
def load_model():
    model_path = 'final_unet_model.pth'
    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    model = UNet(channelExponent=3, dropout=0.).to(device)
    model.load_state_dict(torch.load(model_path))
    return model

In [3]:
%timeit -n 25 -r 25 load_model()

The slowest run took 5.44 times longer than the fastest. This could mean that an intermediate result is being cached.
30.3 ms ± 18.7 ms per loop (mean ± std. dev. of 25 runs, 25 loops each)


In [4]:
model = load_model()
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
image = torch.rand(1, 3, 256, 256).to(device)

def predict_once(image):
    return model(image)

In [5]:
%timeit -n 100 -r 100 predict_once(image)

1.49 ms ± 231 µs per loop (mean ± std. dev. of 100 runs, 100 loops each)


: 

Conclusion:

Model Loading average: 30 ms <br>
Prediction average: 1.5 ms